### ISSUE TITLE
Version Control Data

### ISSUE THREAD

**Issue by awm33** [cognoma/core-service#44]
Track which version of the data (figshare or cancer data sha) that was used for a classifer

**Comment by awm33** [comment]
@dhimmel and/or @cgreene Do you have any thoughts on the best way to handle versioning within the data loader? We currently use `master` from the [cancer-data repo](https://github.com/cognoma/cancer-data) and a hard coded URL for the mutation data stored on figshare.

**Comment by dhimmel** [comment]
The data [from figshare](https://doi.org/10.6084/m9.figshare.3487685) has versions. Therefore, it'd be ideal to specify a version and then download everything we need corresponding to that version. This is what machine-learning [currently does](https://github.com/cognoma/machine-learning/blob/9e0a5273ecd885bd39a1d74f83f648d2eb75da2e/1.download.ipynb).

The `cognoml` package has code for retrieving figshare data ([currently using a class](https://github.com/cognoma/cognoml/blob/81165f4f1caf86a20c17479aa6e2b4d4be242a72/cognoml/data.py), [previously via functions](https://github.com/cognoma/cognoml/blob/f9669edfe0871cd161ffefa3409f67c98dd407a2/cognoml/figshare.py)). We were hoping to move figshare logic to [`cognoma/figshare`](https://github.com/cognoma/figshare) (although we never decided [what exactly to do](https://github.com/cognoma/figshare/issues/1)).

What data is needed from GitHub? We should just upload that to figshare so it can use the common versioning system.

**Comment by cgreene** [comment]
@dhimmel / @gwaygenomics : is this complete? I think that the ml-workers appear to be downloading whatever the latest figshare version is. Does that get reported to the users?

**Comment by dhimmel** [comment]
> Does that get reported to the users?

I don't think it does. I am not sure whether core-service is even storing which figshare version is loaded. The source code for downloading the data is:

https://github.com/cognoma/core-service/blob/b9b2e4f37ac5b250d53ba30c08d94bf38d89c2dd/api/management/commands/acquiredata.py#L21-L39

So it's using the latest from GitHub for all files besides `mutation-matrix.tsv.bz2` in which case it hardlinks to the figshare version 6 file. Instead, I think we should have core service specify a specific figshare version and github commit. If we note the core service commit hash in the output notebook, this would be sufficient to lookup the data versions? (assuming whenever the core-service codebase gets update, the database is reloaded... not sure)

BTW the [figshare](https://doi.org/10.6084/m9.figshare.3487685) has been downloaded 41,471 times. Either people are using this a lot (or more likely we're requesting it an insane number of times :smile_cat:

**Comment by cgreene** [comment]
If we could reconstruct those URLs and put them into the notebook template, that's probably the best way. We'd like users to be able to reproduce the analysis and I think this key ingredient (the exact right data) is missing.
